src/command/archive/get/get.c

```diff
@@ -628,33 +899,67 @@ cmdArchiveGetAsync(void)
                     {
-                        // Get the job and job key
+                        // Get the job
                         ProtocolParallelJob *job = protocolParallelResult(parallelExec);
                         unsigned int processId = protocolParallelJobProcessId(job);
+
+                        // Get wal segment name and archive file map
                         const String *walSegment = varStr(protocolParallelJobKey(job));
+                        const ArchiveFileMap *fileMap = lstFind(checkResult.archiveFileMapList, &walSegment);
+                        ASSERT(fileMap != NULL);
+
+                        // Build warnings for status file
+                        String *warning = strNew("");
+
+                        if (!strLstEmpty(fileMap->warnList))
+                            strCatFmt(warning, "%s", strZ(strLstJoin(fileMap->warnList, "\n")));
 
                         // The job was successful
                         if (protocolParallelJobErrorCode(job) == 0)
                         {
+                            // Get the actual file retrieved
+                            const VariantList *fileResult = varVarLst(protocolParallelJobResult(job));
+                            ArchiveGetFile *file = lstGet(fileMap->actualList, varUIntForce(varLstGet(fileResult, 0)));
+                            ASSERT(file != NULL);
+
+                            // Output file warnings
+                            StringList *fileWarnList = strLstNewVarLst(varVarLst(varLstGet(fileResult, 1)));
+
+                            for (unsigned int warnIdx = 0; warnIdx < strLstSize(fileWarnList); warnIdx++)
+                                LOG_WARN_PID(processId, strZ(strLstGet(fileWarnList, warnIdx)));
+
+                            // Build file warnings for status file
+                            if (!strLstEmpty(fileWarnList))
+                                strCatFmt(warning, "%s%s", strSize(warning) == 0 ? "" : "\n", strZ(strLstJoin(fileWarnList, "\n")));
+
+                            if (strSize(warning) != 0)
+                                archiveAsyncStatusOkWrite(archiveModeGet, walSegment, warning);
+
                             LOG_DETAIL_PID_FMT(
-                                processId,
-                                FOUND_IN_REPO_ARCHIVE_MSG, strZ(walSegment),
-                                cfgOptionGroupIdxToKey(cfgOptGrpRepo, cfgOptionGroupIdxDefault(cfgOptGrpRepo)),
-                                strZ(checkResult.archiveId));
+                                processId, FOUND_IN_REPO_ARCHIVE_MSG, strZ(walSegment),
+                                cfgOptionGroupIdxToKey(cfgOptGrpRepo, file->repoIdx), strZ(file->archiveId));
+
+                            // Rename temp WAL segment to actual name. This is done after the ok file is written so if the ok file
+                            // has warnings it is guaranteed to exist before the foreground process the WAL segment.
```

Comment:
typo: foreground process the --> foreground processes the

Reply:
Actually, was missing "finds". Fixed and also did a bit of rewording in af7bb8faf7ae8ea.